### PR TITLE
RHEL8 release MK.2

### PR DESCRIPTION
**Pull request by evgenyz**
This is a replacement for #121. On top of the changes in that PR we have here:
- Fully functional Packit builds for EPEL8/9
- Test matrix for all Fedoras, with a smoke test.
- FMF/TMT integration for `./tests/smoke.sh` test (former `test_of_the_basic_function.sh`)
- Packit `propose_downstream` integration for `fedora-all`, `epel-9`, `epel-8` (yet to be tested).

### CODE REVIEW

**Comment by evgenyz** [comment]
@matusmarhefka Would this fmf smoke test be enough as a starter for RHEL8/RHEL9? Do we need to adjust the config in some way? Could we be able to re-use the upstream test downstream?

**Comment by matusmarhefka** [comment]
> @matusmarhefka Would this fmf smoke test be enough as a starter for RHEL8/RHEL9? Do we need to adjust the config in some way? Could we be able to re-use the upstream test downstream?

Sure and we can definitely re-use it in downstream. I see we are not testing the EPEL builds, what about testing them on CentOS Stream 8/9? It can be done by updating the job for tests similarly to https://github.com/ComplianceAsCode/content/blob/master/.packit.yaml#L27, so something like this should work:
```
- job: tests
  trigger: pull_request
  identifier: epel-tests
  skip_build: true
  targets:
    - epel-8
        distros: [centos-stream-8]
    - epel-9
        distros: [centos-stream-9]
```

File: .packit.yaml [hunk 1]
```diff
@@ -31,15 +35,32 @@ jobs:
 - job: propose_downstream
   trigger: release
   dist_git_branches:
-    - fedora-rawhide
+    - fedora-all
```
Comment (Honny1) [inline .packit.yaml:38]: Specifying all possible branches downstream is not good. PackIt will create a PR per branch and they will all be similar. An exception might be EPEL 8. I think EPEL 8 will need a separate job to create a PR with a spec file for RHEL8.

A subsequent fedora-release is done by merging the PR created by PackIt into the rawhide(main) branch, and all other branches are merged into rawhide(main). So all Fedoras and EPEL9 versions have the same content. Creating multiple PRs with the same content is not necessary.

**Comment by Honny1** [comment]
@evgenyz I think the `epel-*` tests (including Fedora) will install the released package. The problem is probably caused by `skip_build: true`. The test uses the available released package. So it does not use the RPMs created in `copr`, so the change in PR will not be propagated in the RPMs used. And `epel-8` fails because `openscap-report` is not available in the EPEL8 repository.


This row in `epel-9` [log](https://artifacts.dev.testing-farm.io/da1156be-1876-4739-a88a-005e709b2912/work-smoke8obd9xst/log.txt) is proof.
```
10:09:09             out:  openscap-report       noarch  0.2.1-1.el9   epel                         569 k
```

**Comment by evgenyz** [comment]
Okay, the way Packit see it, if `skip_build: true` is provided then package installation is supposed to be part of `tmt` definition. The `skip_build: true` config is a workaround for a completely different use-case, when the package is coming from a compose (or somewhere else rather than packit build process).

I propose to kick out the `epel-8` config for Packit until they get rid of implicit builds for tests job. I suppose that `package_dependency_test` is enough for upstream testing.

**Pull request merged**